runelite-client/src/main/java/net/runelite/client/plugins/zalcano/ZalcanoOverlay.java

```diff
@@ -0,0 +1,263 @@
+import net.runelite.api.GameObject;
+import net.runelite.api.GraphicsObject;
+import net.runelite.api.Perspective;
+import net.runelite.api.Renderable;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+/**
+ * some ids
+ * 9049 = zalcano npc id 8439 wakeup animation 8437 mineable stage
+ * 1727 = rocks falling graphics object id
+ * 1728 = red projectile flying into rock
+ * <p>
+ * 36192 + anim 8448 = glowing rock
+ * <p>
+ * 36199 red circle
+ * 36200 blue circle
+ * <p>
+ * golem id = 9051
+ * <p>
+ * 23905, 23906, 23907
+ */
+@Slf4j
+public class ZalcanoOverlay extends Overlay
+{
+
+	private final ZalcanoPlugin plugin;
+	private final ZalcanoConfig config;
+	private final ZalcanoUtil util;
+	private final Client client;
+
+
+	@Inject
+	ZalcanoOverlay(final ZalcanoPlugin plugin, final ZalcanoConfig config, final ZalcanoUtil util, final Client client)
+	{
+		super(plugin);
+		this.plugin = plugin;
+		this.config = config;
+		this.util = util;
+		this.client = client;
+
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.HIGH);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (config.showAoeForRockfall())
+		{
+			renderBadTiles(graphics);
+		}
+		if (config.showAoeForRedSymbols())
+		{
+			renderRedSymbols(graphics);
+		}
+		if (config.highlightMiningSpot())
+		{
+			renderRockToMine(graphics);
+		}
+		if (config.highlightGolem())
+		{
+			renderGolem(graphics);
+		}
+		if (config.highlightZalcanoHull())
+		{
+			renderZalcano(graphics);
+		}
+
+		//has their own configs within this method
+		renderZalcanoAnimations(graphics);
+
+		return null;
+	}
+
+	private void renderBadTiles(Graphics2D graphics)
+	{
+		ArrayList<GraphicsObject> arrList = new ArrayList<>();
+		for (GraphicsObject graphicsObject : client.getGraphicsObjects())
+		{
+			if (graphicsObject.getId() == 1727)
+			{
+				arrList.add(graphicsObject);
+			}
+		}
+
+		for (GraphicsObject graphicsObject : arrList)
+		{
+			WorldPoint worldPoint = WorldPoint.fromLocal(client, graphicsObject.getLocation());
+			OverlayUtil.drawTiles(graphics, client, worldPoint, client.getLocalPlayer().getWorldLocation(), Color.RED, 2, 150, 50);
+		}
+	}
+
+	private void renderRedSymbols(Graphics2D graphics)
+	{
+		for (GameObject gameObject : util.getGameObjects())
+		{
+			if (gameObject.getId() == 36199)
+			{
+				if (client.getLocalPlayer().getLocalLocation().distanceTo(gameObject.getLocalLocation()) <= 2400)
+				{
+					Renderable renderable = gameObject.getRenderable();
+					if (renderable instanceof DynamicObject)
```

Comment:
Does it need to be a dynamic object to render the overlay?

Reply:
In this case yes